File: packages/ios/Byoky/Views/Wallet/OfflineUpgradeBanner.swift
import SwiftUI

private let dismissCooldown: TimeInterval = 7 * 24 * 3600

struct OfflineUpgradeBanner: View {
    @EnvironmentObject var wallet: WalletStore
    let onActivate: () -> Void

    var body: some View {
        if shouldShow {
            VStack(alignment: .leading, spacing: 8) {
                HStack(alignment: .top) {
                    VStack(alignment: .leading, spacing: 4) {
                        Text("Sync across devices")
                            .font(.subheadline.bold())
                            .foregroundStyle(Theme.textPrimary)
                        Text("Turn on Cloud Sync to access your keys on any device, end-to-end encrypted.")
                            .font(.caption)
                            .foregroundStyle(Theme.textSecondary)
                            .fixedSize(horizontal: false, vertical: true)
                    }
                    Spacer()
                    Button {
                        wallet.dismissVaultBanner()
                    } label: {
                        Image(systemName: "xmark")
                            .font(.caption)
                            .foregroundStyle(Theme.textMuted)
                    }
                }

                Button {
                    onActivate()
                } label: {
                    Text("Activate Cloud Sync")
                        .font(.caption.bold())
                        .foregroundStyle(.white)
                        .padding(.horizontal, 12)
                        .padding(.vertical, 6)
                        .background(Theme.accent)
                        .clipShape(RoundedRectangle(cornerRadius: 6))
                }
            }
            .padding(12)
            .background(Theme.accent.opacity(0.08))
            .clipShape(RoundedRectangle(cornerRadius: 10))
            .overlay(
                RoundedRectangle(cornerRadius: 10)
                    .strokeBorder(Theme.accent, lineWidth: 1)
            )
            .padding(.bottom, 12)
        }
    }

    private var shouldShow: Bool {
        if wallet.cloudVaultEnabled { return false }
        if let dismissed = wallet.vaultBannerDismissedAt, Date().timeIntervalSince(dismissed) < dismissCooldown {
            return false
        }
        return true
    }
}

File: packages/android/app/src/main/java/com/byoky/app/ui/components/OfflineUpgradeBanner.kt
package com.byoky.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.byoky.app.data.WalletStore
import com.byoky.app.ui.theme.*

private const val DISMISS_COOLDOWN_MS = 7L * 24 * 60 * 60 * 1000

@Composable
fun OfflineUpgradeBanner(wallet: WalletStore, onActivate: () -> Unit) {
    val cloudVaultEnabled by wallet.cloudVaultEnabled.collectAsState()
    val dismissedAt by wallet.vaultBannerDismissedAt.collectAsState()

    val shouldShow = !cloudVaultEnabled && (dismissedAt == 0L || System.currentTimeMillis() - dismissedAt > DISMISS_COOLDOWN_MS)
    if (!shouldShow) return

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(Accent.copy(alpha = 0.08f), RoundedCornerShape(10.dp))
            .border(1.dp, Accent, RoundedCornerShape(10.dp))
            .padding(12.dp),
    ) {
        Row(verticalAlignment = Alignment.Top) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    "Sync across devices",
                    fontSize = 13.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = TextPrimary,
                )
                Spacer(Modifier.height(4.dp))
                Text(
                    "Turn on Cloud Sync to access your keys on any device, end-to-end encrypted.",
                    fontSize = 12.sp,
                    color = TextSecondary,
                )
            }
            IconButton(onClick = { wallet.dismissVaultBanner() }, modifier = Modifier.size(24.dp)) {
                Icon(Icons.Default.Close, contentDescription = "Dismiss", tint = TextMuted, modifier = Modifier.size(14.dp))
            }
        }

        Spacer(Modifier.height(8.dp))

        Button(
            onClick = onActivate,
            shape = RoundedCornerShape(6.dp),
            colors = ButtonDefaults.buttonColors(containerColor = Accent),
            contentPadding = PaddingValues(horizontal = 12.dp, vertical = 6.dp),
        ) {
            Text("Activate Cloud Sync", fontSize = 12.sp, fontWeight = FontWeight.SemiBold)
        }
    }
}
